### PR TITLE
[12.0][MIG] web_tree_dynamic_colored_field

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Standard migration.

### CODE REVIEW

File: web_tree_dynamic_colored_field/README.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+========
+
+* Add attribute ``bg_color`` on field's ``options`` to color background of a cell in tree view
+* Add attribute ``fg_color`` on field's ``options`` to change text color of a cell in tree view
+* Add attribute ``color_field`` on the tree element's ``colors`` to use as color
+
+Usage
+=====
+
+* In the tree view declaration, put ``options='{"bg_color": "red: customer==True"}`` attribute in the ``field`` tag::
```
Comment (sudhir-erpharbor) [inline web_tree_dynamic_colored_field/README.rst:28]: Please also add an example to use char type of field (selection) in the condition.
Ex: options='{"bg_color": "red: state == 'rejected'' "}

File: web_tree_dynamic_colored_field/README.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    With this example, column which renders 'name' field will have its text colored in white on a customer records.
+
+* In the tree view declaration, use ``options='"color_field": "my_color"'`` attribute in the ``tree`` tag::
+
+    ...
+    <field name="arch" type="xml">
+        <tree string="View name" colors="color_field: my_color" >
+            ...
+            <field name="my_color" invisible="1"/>
```
Comment (sudhir-erpharbor) [inline web_tree_dynamic_colored_field/README.rst:62]: Can we use HEX color codes in this field or it should always be the color name?
Correct me if I understood it in wrong way.

**Review by sudhir-erpharbor — CHANGES_REQUESTED** [review]
Minor suggestions

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
Please, check my comment below

File: web_tree_dynamic_colored_field/demo/res_users.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        <field name="model">res.users</field>
+        <field name="inherit_id" ref="base.view_users_tree" />
+        <field name="arch" type="xml">
+          <tree position="attributes">
+            <attribute name="colors">color_field: lang</attribute>
+          </tree>
+          <field name="login_date" position="attributes">
+            <attribute name="options">{
+              "bg_color": "#9e1635: login_date == False",
+              "fg_color": "white: login_date == False"
```
Comment (LoisRForgeFlow) [inline web_tree_dynamic_colored_field/demo/res_users.xml:13]: I just tested locally and this view is installed with demo but no change to the background color happened:
image

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]

File: web_tree_dynamic_colored_field/__manifest__.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
     'name': 'Colorize field in tree views',
     'summary': 'Allows you to dynamically color fields on tree views',
     'category': 'Hidden/Dependency',
-    'version': '11.0.1.0.1',
+    'version': '12.0.1.0.1',
```
Comment (LoisRForgeFlow) [inline web_tree_dynamic_colored_field/__manifest__.py]: 12.0.1.0.0

File: web_tree_dynamic_colored_field/README.rst (hunk 2 above)
Comment (MiquelRForgeFlow) [reply web_tree_dynamic_colored_field/README.rst:62]: You can see in the demo data that the HEX colors also can be used.

File: web_tree_dynamic_colored_field/README.rst (hunk 1 above)
Comment (sudhir-erpharbor) [reply web_tree_dynamic_colored_field/README.rst:28]: Any clue on this?
@mreficent

**Comment by oerp-odoo** [comment]
Hi,
Is this module migration already finished, or there are some issues with it, that PR is not yet merged?

**Review by LoisRForgeFlow — APPROVED** [review]
:+1:

**Review by HviorForgeFlow — APPROVED** [review]
Functional Review :+1:

**Pull request merged**